pkg/queue/manager.go

```diff
@@ -296,7 +305,34 @@ func (m *Manager) deleteWorkloadFromQueueAndClusterQueue(w *kueue.QueuedWorkload
 	cq := m.clusterQueues[q.ClusterQueue]
 	if cq != nil {
 		cq.Delete(w)
+		if w.Spec.Admission != nil && m.queueAllInadmissibleWorkloadsInCohort(cq) {
+			m.cond.Broadcast()
+		}
+	}
+}
+
+// queueAllInadmissibleWorkloadsInCohort moves all workloads in the same
+// cohort with this ClusterQueue from inadmissibleWorkloads to heap. If the
+// cohort of this ClusterQueue is empty, it just moves all workloads in this
+// ClusterQueue. If at least one workload is moved, returns true. Otherwise
+// returns false.
+// The events listed below could make workloads in the same cohort admissible.
+// Then queueAllInadmissibleWorkloadsInCohort need to be invoked.
+// 1. delete events for any admitted workload in the cohort.
+// 2. add events of any cluster queue in the cohort.
+// 3. update events of any cluster queue in the cohort.
+func (m *Manager) queueAllInadmissibleWorkloadsInCohort(cq ClusterQueue) bool {
+	if len(cq.Cohort()) == 0 {
+		return cq.QueueAllInadmissibleWorkloads()
+	}
+
+	queued := false
+	for _, c := range m.clusterQueues {
```

Comment:
We should try to do better than that by having a cohort to ClusterQueues index.

Reply:
Done.